github/Organization.py

```diff
@@ -582,6 +582,47 @@ def create_repo(
             self._requester, headers, data, completed=True
         )
 
+    def create_secret(
+        self,
+        secret_name,
+        unencrypted_value,
+        visibility="all",
+        selected_repositories=[],
+    ):
+        """
+        :calls: `PUT /orgs/{org}/actions/secrets/{secret_name} <https://docs.github.com/en/rest/reference/actions#create-or-update-an-organization-secret>`_
+        :param secret_name: string
+        :param unencrypted_value: string
+        :param visibility: string
+        :param selected_repositories: list of :class:`github.Repository.Repository`
+        :rtype: bool
+        """
+        assert isinstance(secret_name, str), secret_name
+        assert isinstance(unencrypted_value, str), unencrypted_value
+        assert isinstance(visibility, str), visibility
+        if visibility == "selected":
+            assert isinstance(selected_repositories, list) and all(
+                isinstance(element, github.Repository.Repository)
+                for element in selected_repositories
+            ), selected_repositories
+
+        public_key = self.get_public_key()
+        payload = public_key.encrypt(unencrypted_value)
+        put_parameters = {
+            "key_id": public_key.key_id,
+            "encrypted_value": payload,
+            "visibility": visibility,
+        }
+        if selected_repositories:
```

Comment:
This isn't guarded like the assert check, which means that if you don't pass visibility and do pass selected_repositories something that isn't a list of Repository objects, then we throw an exception.

Do we need to be more careful here and assert that it's an empy list (or NotSet) if visibility isn't selected?

Reply:
yeah, it this was not covered, thanks
Made changes to handle cases:
visibility==selected and selected_repositories is NotSet
visibility!=selected and selected_repositories is not NotSet

@s-t-e-v-e-n-k Please check it